transport.go

```diff
@@ -186,6 +186,7 @@ func (t *transport) Dial(ctx context.Context, raddr ma.Multiaddr, p peer.ID) (tp
 
 	connaddrs := &connAddrs{lmAddr: localMultiaddr, rmAddr: remoteMultiaddr}
 	if t.gater != nil && !t.gater.InterceptSecured(n.DirOutbound, p, connaddrs) {
+		sess.CloseWithError(0, "")
```

Comment:
We *could* use a different error code and / or a reason string here. Might help debugging, but then this would be something QUIC-specific as afaik we don't have any close reasons for TCP-based transports.